casper/src/main/scala/coop/rchain/casper/MultiParentCasperImpl.scala

```diff
@@ -176,14 +176,34 @@ class MultiParentCasperImpl[F[_]: Sync: Concurrent: Sync: ConnectionsCell: Trans
    */
   def createBlock: F[CreateBlockStatus] = validatorId match {
     case Some(ValidatorIdentity(publicKey, privateKey, sigAlgorithm)) =>
+      def updateDeployValidAfterBlock(deployData: DeployData, max: Long) =
+        if (deployData.validAfterBlockNumber == -1)
+          deployData.withValidAfterBlockNumber(max)
+        else
+          deployData
+
+      def updateDeployHistory(state: CasperState, max: Long) =
+        state.copy(deployHistory = state.deployHistory.map(deployData => {
+          updateDeployValidAfterBlock(deployData, max)
+        }))
+
       for {
         dag       <- blockDag
         tipHashes <- estimator(dag)
         p         <- chooseNonConflicting[F](tipHashes, dag)
         _ <- Log[F].info(
               s"${p.size} parents out of ${tipHashes.size} latest blocks will be used."
             )
-        maxBlockNumber   = ProtoUtil.maxBlockNumber(p)
+        maxBlockNumber = ProtoUtil.maxBlockNumber(p)
+        /*
+         * This mechanism is a first effort to make life of a deploying party easier.
+         * Instead of expecting the user to guess the current block number we assume that
+         * if no value is given (default: -1) rchain should try to deploy
+         * with the current known max block number.
+         *
+         * TODO make more developer friendly by introducing Option instead of a magic number
+         */
+        _                <- Cell[F, CasperState].modify(state => updateDeployHistory(state, maxBlockNumber))
```

Comment:
I've got very superficial non-understanding of the issue at hand, but wouldn't having a default offset from the given node's known maxBlockNumber make things better?

If I give a value bigger than my known max block - I'm proposing an invalid block according to my knowledge?

If I give a value bigger than other node's max block number - will they consider the block invalid until they get it from the other node? Maybe I should propose a smaller number?